### ISSUE TITLE
Error stating alias already specified as version, but it is not

### ISSUE THREAD

**Issue by robzor92** [jimporter/mike#112]
Hello! First of all, thanks a bunch for this tool. It really helped out with versioning documentation for us.

Recently we've run into an issue with this command failing.

 `mike deploy 3.0.1 3.0 -u --push`

with the output:

`error: alias '3.0' already specified as a version`

We have a setup with an alias named `3.0` that points to the latest release version, in this case `3.0.1`. There is no version named `3.0` as the error states. All our versions are specified as `*.*.*`. 

Also, in addition to the `3.0` alias, we have another alias named `latest` which just points to the overall latest version.

Our versions.json looks like this:

```
[
   {
      "version":"3.1.0-SNAPSHOT",
      "title":"3.1.0-SNAPSHOT",
      "aliases":[
         "dev"
      ]
   },
   {
      "version":"3.0.0",
      "title":"3.0.0",
      "aliases":[
         "3.0",
         "latest"
      ]
   },
   {
      "version":"2.5.2",
      "title":"2.5.2",
      "aliases":[
         
      ]
   },
   {
      "version":"2.5.1",
      "title":"2.5.1",
      "aliases":[
         
      ]
   },
   {
      "version":"2.5.0",
      "title":"2.5.0",
      "aliases":[
         
      ]
   }
]
```
Here is our [gh-pages](https://github.com/logicalclocks/machine-learning-api/tree/gh-pages) branch.

Are we doing something wrong on our end?

Grateful for any help on this!

**Comment by jimporter** [comment]
The issue here was that `3.0` and `3.0.0` were getting compared to each other as *versions*, not as *strings*. As versions, those two are the same, which resulted in this false positive.

**Comment by robzor92** [comment]
@jimporter Thanks a lot! Happy to try it out when 2.0.0 is released. :)

**Comment by moritzmeister** [comment]
@jimporter thanks for the quick response! Do you have plans for a minor bug fix release in the near future?

**Comment by jimporter** [comment]
No, this will be released with 2.0, which will be tagged when I close out everything left in the [milestone](https://github.com/jimporter/mike/milestone/1). That probably won't be particularly soon though, since this project is a low priority for me, and I don't have a lot of free time lately.

If you need this sooner, I'd recommend using the development builds. They should be pretty stable, and more testing is always appreciated.
